pkg/ipmi/ocp/set_vpd.go

```diff
@@ -10,15 +10,25 @@ import (
 	"log"
 	"os"
 	"os/exec"
+	"os/signal"
+	"syscall"
 )
 
+func handler(c <-chan os.Signal) {
+	<-c
+	log.Printf("ignoring SIGINT during flash write to prevent corruption")
+}
+
 // Set RW_VPD key-value
 func Set(key string, value []byte) error {
 	file, err := ioutil.TempFile("/tmp", "rwvpd*.bin")
 	if err != nil {
 		return err
 	}
+	c := make(chan os.Signal)
 	defer os.Remove(file.Name())
+	defer signal.Reset(syscall.SIGINT)
+	go handler(c)
```

Comment:
We need to close c so the handler loop can exit. This must strictly be done after the signal.Reset or we can get a panic for send on a closed channel. So let's setup the handler, defer the channel close, and then later stack the signal.Reset defer on top. (also keep the file defer with the temp file where it belongs)
```suggestion
	defer os.Remove(file.Name())
	
	c := make(chan os.Signal)
	go handler(c)
	defer close(c)
	// move defer signal.Reset down to below the signal.Notify
```

Reply:
Thanks for the suggestion, done.